packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -694,8 +704,19 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             },
         );
 
+        // back-compat before namespaces were improved
+        let dataStoresSnapshot = context.baseSnapshot;
+        let dataStoresSnapshotType: BaseSnapshotType = "legacy";
+
+        if (!!dataStoresSnapshot && metadata.snapshotFormatVersion !== containerSnapshotFormatVersions.missing) {
```

Comment:
I'm a bit at a loss in terms of using "missing" & "next". What if we need to change format again?
Would approach of just saying that existing format is 1.0 (and that's default value if attributes are missing) and new format is 2.0 work? Then we can clearly articulate what is 1.0 format, what is 2.0 format, write documentation, and someone willing to write code on the side to parse our files can actually do it.
Maybe that's what I'm missing - formal description of format, i.e. what is the instruction to me as a side developer of how to read and write these formats, without looking in the code to understand formats and expectations.

Reply:
I see - you essentially have 3 versions. I'd rather call them 1.0, 2.0 and 3.0 and make 1.0 default - a bit cleaner IMHO

Reply:
This makes sense to me, I was trying to not diverge as far from the pre-existing "current" logic, but I agree it probably makes more sense not to use sliding version names when reading.